projects/storefrontlib/src/shared/components/form/form-errors/form-errors.component.ts

```diff
@@ -1,36 +1,35 @@
-import { Component, Input } from '@angular/core';
+import {
+  Component,
+  Input,
+  ChangeDetectionStrategy,
+  OnInit,
+} from '@angular/core';
 import { FormControl } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { startWith, map, filter, throttleTime } from 'rxjs/operators';
 
 /**
  * This component renders form errors.
  */
 @Component({
   selector: 'cx-form-errors',
   templateUrl: './form-errors.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class FormErrorComponent {
+export class FormErrorComponent implements OnInit {
   @Input() control: FormControl;
+  errors$: Observable<string[]>;
 
-  /**
-   * Checks if the control is invalid (contains any validation errors).
-   * Also checks if the control was already touched (either with click or focus).
-   * @returns boolean
-   */
-  shouldShowErrors(): boolean {
-    return this.control.invalid && (this.control.dirty || this.control.touched);
-  }
-
-  /**
-   * Retrieves error names from control's errors array.
-   * @returns string[] List of errors' names
-   */
-  getErrorNames(): string[] {
-    if (this.shouldShowErrors() && this?.control?.errors) {
-      return Object.entries(this.control.errors)
-        .filter(error => error[1])
-        .map(error => error[0]);
-    }
-
-    return [];
+  ngOnInit() {
+    this.errors$ = this.control.statusChanges.pipe(
+      startWith({}),
+      throttleTime(500),
+      filter(() => !!this?.control?.errors),
+      map(() =>
+        Object.entries(this.control.errors)
```

Comment:
hard to read. I believe we can simplify to:
```typescript
 this.errors$ = this.control.statusChanges.pipe(
    startWith({}),
    map(() => this.control.errors || {}),
    map(errors => Object.keys(errors))
);
```

Reply:
partially fixed - `Object.entries()` needs to be used (with the following logic), because we need to check which validation errors have `true` - we cannot just return all errors